pennylane/devices/tests/test_gates.py

```diff
@@ -150,6 +204,49 @@
 # ===============================================================
 
 
+class TestSupportedGates:
+    """Test that the device can implement all gates that it claims to support."""
+
+    @pytest.mark.parametrize("operation", all_ops)
+    def test_supported_gates_can_be_implemented(self, device_kwargs, operation):
+        """Test that the device can implement all its supported gates."""
+        device_kwargs["wires"] = 3  # maximum size of current gates
+        dev = qml.device(**device_kwargs)
+
+        assert hasattr(dev, "operations")
+        if operation in dev.operations:
+
+            @qml.qnode(dev)
+            def circuit():
+                ops[operation]
+                return qml.expval(qml.Identity(wires=0))
+
+            assert isinstance(circuit(), (float, np.ndarray))
```

Comment:
Just curious, what would be a specific case for this assertion to fail?

Reply:
Good question, the assert statement is more of a dummy here. It's one of those tests where I just want to know "does the execution throw an error". Is there a generic way to do this with pytest? One could also not have an assert statement at all, as in some other test, but I thought this is bad practice...